applets/clock/clock.c

```diff
@@ -888,105 +893,181 @@ create_calendar (ClockData *cd)
+                gdk_window_get_geometry (panelwin, NULL, NULL, &panel_w, &panel_h);
+
+                /*Set up GTK Layer Shell*/
+                gtk_layer_init_for_window (window);
+                gtk_layer_set_layer (window, GTK_LAYER_SHELL_LAYER_TOP);
+
+                switch (cd->orient) {
+                case MATE_PANEL_APPLET_ORIENT_RIGHT:
+                        gtk_layer_set_anchor (window, GTK_LAYER_SHELL_EDGE_LEFT, TRUE);
+                        if (y < (panel_h -h))
+                        {
+                                gtk_layer_set_anchor (window, GTK_LAYER_SHELL_EDGE_TOP, TRUE);
+                                gtk_layer_set_margin (window, GTK_LAYER_SHELL_EDGE_TOP, y);
+                        }
+                        else
+                                gtk_layer_set_anchor (window, GTK_LAYER_SHELL_EDGE_BOTTOM, TRUE);
 
-                gravity = GDK_GRAVITY_SOUTH_WEST;
+                        break;
+                case MATE_PANEL_APPLET_ORIENT_LEFT:
+                        gtk_layer_set_anchor (window, GTK_LAYER_SHELL_EDGE_RIGHT, TRUE);
+                        if (y < (panel_h - h))
+                        {
+                                gtk_layer_set_anchor (window, GTK_LAYER_SHELL_EDGE_TOP, TRUE);
+                                gtk_layer_set_margin (window, GTK_LAYER_SHELL_EDGE_TOP, y);
+                        }
+                        else
+                                gtk_layer_set_anchor (window, GTK_LAYER_SHELL_EDGE_BOTTOM, TRUE);
 
-                break;
-        }
+                        break;
+                case MATE_PANEL_APPLET_ORIENT_DOWN:
+                        gtk_layer_set_anchor (window, GTK_LAYER_SHELL_EDGE_TOP, TRUE);
+                        if (x < (panel_w -w))
+                        {
+                                gtk_layer_set_anchor (window, GTK_LAYER_SHELL_EDGE_LEFT, TRUE);
+                                gtk_layer_set_margin (window, GTK_LAYER_SHELL_EDGE_LEFT, x);
+                        }
+                        else
+                                gtk_layer_set_anchor(window, GTK_LAYER_SHELL_EDGE_RIGHT, TRUE);
+
+                        break;
+                case MATE_PANEL_APPLET_ORIENT_UP:
+                        gtk_layer_set_anchor (window, GTK_LAYER_SHELL_EDGE_BOTTOM, TRUE);
+                        if (x < (panel_w -w))
+                        {
+                                gtk_layer_set_anchor (window, GTK_LAYER_SHELL_EDGE_LEFT, TRUE);
+                                gtk_layer_set_margin (window, GTK_LAYER_SHELL_EDGE_LEFT, x);
+                        }
+                        else
+                                gtk_layer_set_anchor (window, GTK_LAYER_SHELL_EDGE_RIGHT, TRUE);
 
-        gtk_window_move (GTK_WINDOW (cd->calendar_popup), x, y);
-        gtk_window_set_gravity (GTK_WINDOW (cd->calendar_popup), gravity);
+                        break;
+                }
+                return;
+        }
```

Comment:
Line 1070 starts with a tab.

Reply:
No tab found here, select in in browser here or in the file and you can move the cursor to each position, with a tab it jumps. That is not true with tabs. Not sure what's going on with that

Reply:
This is weird, when look in clock.c local in your branch i see no tab.
If there is really a mysterious tab after merging we can fix it later.
No need to spend time on it.